src/main/kotlin/com/cultureamp/eventsourcing/RelationalDatabaseEventStore.kt

```diff
@@ -256,36 +247,57 @@ object H2DatabaseEventStore {
         noinline afterSinkHook: (List<SequencedEvent<M>>) -> Unit,
         eventsSinkTableName: String,
     ): RelationalDatabaseEventStore<M> {
-        return RelationalDatabaseEventStore(db, eventsTable(tableName), EventsSequenceStats(eventsSequenceStateTableName), M::class.java, objectMapper, eventTypeResolver, afterSinkHook = afterSinkHook, eventsSinkTable = Events(eventsSinkTableName, Table::jsonb), )
+        return RelationalDatabaseEventStore(
+            db,
+            Events(tableName, objectMapper, object : TypeReference<M>() {}),
+            EventsSequenceStats(eventsSequenceStateTableName),
+            objectMapper,
+            eventTypeResolver,
+            afterSinkHook = afterSinkHook,
+            eventsSinkTable = Events(eventsSinkTableName, objectMapper, object : TypeReference<M>() {}),
+        )
     }
-
-    @PublishedApi
-    internal fun eventsTable(tableName: String = defaultEventsTableName) = Events(tableName) { name -> this.text(name) }
 }
 
 internal fun <T> String.asClass(): Class<out T>? {
     @Suppress("UNCHECKED_CAST")
     return Class.forName(this) as Class<out T>?
 }
 
-class Events(tableName: String = defaultEventsTableName, jsonb: Table.(String) -> Column<String> = Table::jsonb) :
-    Table(tableName) {
+class Events<M : EventMetadata>(tableName: String = defaultEventsTableName, private val objectMapper: ObjectMapper = defaultObjectMapper, metadataType: TypeReference<M>) : Table(tableName) {
```

Comment:
It feels mega wierd to have an objectMapper on a Table class. Is there a reason we need to use the in-built json? I like the old way better TBH! That said there's nothing show-stoppy about it, just feels a little unclean

Reply:
hmm yeah I do kinda feel this too - the custom JsonB column we had stopped working with the Exposed bump, and I fiddled around for a while trying to get it working and didn't quite get there. Probably with enough time I could get that working, but it was sort of a signal to me that we probably should just use the one that ships with Exposed because it will be supported by them and continue to work. I'm pretty open to either though, tbh, do you think it's worth my spending another hour trying to get the custom JsonB column working?